torrentinfo.py

```diff
@@ -2,37 +2,81 @@
 torrentinfo.py - Sopel module to fetch torrent information for links sent to channels
 """
 
+# -*- coding: utf-8 -*-
+# encoding=utf8  
+import sys  
+reload(sys)  
+sys.setdefaultencoding('utf-8')
+
 from sopel import module
-import bleach
 import requests
 from lxml import etree
+from pyquery import PyQuery
+import re
 
-NYAA_URL = 'https://www.nyaa.se/?page=view&tid=%s'
-
+NYAA_URL = 'https://nyaa.si/view/%s'
+ANIDEX_URL = 'https://anidex.info/torrent/%s'
 
-@module.rule('.*https?:\/\/(?:www\.)?nyaa.(?:se|eu)\/\S*(?:\?|&)tid=(\d+).*')
+@module.rule('.*https?:\/\/(?:www\.)?nyaa\.si\/(?:view|download)\/(\d+).*')
 def nyaa_info(bot, trigger):
+    parsed_url = NYAA_URL % trigger.group(1)
+    try:
+        r = requests.get(url=parsed_url, timeout=(10.0, 4.0))
+    except requests.exceptions.ConnectTimeout:
+        returnbot.say("Connection timed out.")
+    except requests.exceptions.ConnectionError:
+        return bot.say("Couldn't connect to server.")
+    except requests.exceptions.ReadTimeout:
+        return bot.say("Server took too long to send data.")
+    try:
+        r.raise_for_status()
+    except requests.exceptions.HTTPError as e:
+        return bot.say("[Nyaa] HTTP error: " + e.message)
+
+    page = etree.HTML(r.content)
+    pq = PyQuery(page)
+
+    t = {}
+
+    t['name'] = pq('body > div > div:nth-child(1) > div.panel-heading > h3')
+    t['size'] = pq('body > div > div:nth-child(1) > div.panel-body > div:nth-child(4) > div:nth-child(2)')
+    t['uploader'] = pq('body > div > div:nth-child(1) > div.panel-body > div:nth-child(2) > div:nth-child(2) > a')
+    t['link'] = parsed_url
+    t['name'] = t['name'].text()
+    t['size'] = t['size'].text()
+    t['uploader'] = t['uploader'].text()
```

Comment:
These `text()` calls can be done inline above. If the selectors can be simplified, all the better. (Is `:nth-child` really the only way to get at the needed elements? That's really vulnerable to even slight site changes. :neutral_face:)

Reply:
Seems like it is, as there are multiple .row and and multiple .col-md-5 in .panel-body.